### DISCUSSION TITLE
500 on Server Action POST

### DISCUSSION THREAD

**Discussion by neilck** [vercel/next.js#62918] [App Router] [answered]
Anyone know why server action causes 500 on POST only in production?

I've been trying to use Server Actions, but keep getting "Failed to load resource: the server responded with a status of 500 ()" error. I think it's the POST back to the page after server action has run.

To troubleshoot, I created a simple example of the code which demonstrates this error.

src/app/test/actions.tsx
```typescript
"use server";

export async function sayHello() {
  console.log("sayHello called");
  return "Hello";
}
````

src/app/test/page.tsx
````typescript
"use client";

import { useState, useEffect } from "react";
import { sayHello } from "./actions";

export default function Page() {
  const [message, SetMessage] = useState("goodbye");

  useEffect(() => {
    const callSayHello = async () => {
      const mesg = await sayHello();
      SetMessage(mesg);
    };

    callSayHello();
  });
  return (
    <>
      <p>{message}</p>
    </>
  );
}
````

package.json dependencies
```
  "dependencies": {
    "@emotion/cache": "^11.11.0",
    "@emotion/react": "^11.11.1",
    "@emotion/styled": "^11.11.0",
    "@mui/material": "^5.15.11",
    "debug": "^4.3.4",
    "next": "^14.1.2",
    "next-recaptcha-v3": "^1.1.5",
    "react": "^18",
    "react-dom": "^18",
    "react-google-recaptcha": "^2.1.0"
  }
```

Its a Next 14 app router app, with a custom domain set in Vercel, on the Hobby plan.

Any help is appreciated.

**Comment by neilck** [comment]
Here's the error
![image](https://github.com/vercel/next.js/assets/11378702/401834a6-bd50-4a1a-9cda-fe7823ce5417)

**Comment by MH4GF** [comment]
I'm facing the same problem. Here is the minimum repro:

- https://debug-server-actions.vercel.app/
- https://github.com/MH4GF/debug-server-actions
    - server action: https://github.com/MH4GF/debug-server-actions/blob/d6b7fdbf840819bc61d073f2544ec86b66f00ca3/src/app/sampleAction.ts

But, server action log returns 200.

<img width="962" alt="スクリーンショット 2024-03-06 19 32 24" src="https://github.com/vercel/next.js/assets/31152321/829e9c79-26f9-47b4-b3e5-b60b284302f9">

**Reply by icyJoseph** [reply]
I think you have enough there to create an Issue, if you haven't already.

**Reply by MH4GF** [reply]
I associated it here: https://github.com/vercel/next.js/issues/62932

**Reply by neilck** [reply]
Thanks!

**Reply by devjiwonchoi** [reply]
This repro is not throwing 500. How is it related to this issue?

**Comment by neilck** [comment, marked as answer]
Vercel seems to a have fixed this, with no changes from me.
![image](https://github.com/vercel/next.js/assets/11378702/269025cd-2b70-4792-823a-ba634e6fc4a8)

**Reply by eriknyk** [reply]
@neilck  but did you update a a specific nextjs version? what version?

**Reply by neilck** [reply]
Same version as originally posted 14.1
